### PR TITLE
Tracking Opt In in Discussions

### PR DESCRIPTION
**Pull request by nikodamn**
Use already implemented TrackingOptInModal page object and mobile-wiki tests cases in Discussions. Add check for anonymized request to Google Analytics.

@Wikia/iris @ludwikkazmierczak  @bulaszenko @chessky

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L58](https://github.com/Wikia/selenium-tests/blob/b4cff7ec7508aefdca7c97b3af861172cb784283/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L58): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L167](https://github.com/Wikia/selenium-tests/blob/b4cff7ec7508aefdca7c97b3af861172cb784283/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L167): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L176](https://github.com/Wikia/selenium-tests/blob/b4cff7ec7508aefdca7c97b3af861172cb784283/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L176): Introduce a new variable instead of reusing the parameter "url". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/common/dataprovider/TrackingOptInDataProvider.java [hunk 1]
```diff
@@ -71,6 +69,10 @@
           "wgAdDriverA9VideoBidderCountries"
   };
 
+  private static final String DISCUSSION_GOOGLE_ANONYMIZED =
```
Comment (rogatty) [inline src/test/java/com/wikia/webdriver/common/dataprovider/TrackingOptInDataProvider.java]: This isn't exclusive to Discussions.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L58](https://github.com/Wikia/selenium-tests/blob/3e4c31502c145b3a757f0c6d138b0ed9a9dfaa94/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L58): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L167](https://github.com/Wikia/selenium-tests/blob/3e4c31502c145b3a757f0c6d138b0ed9a9dfaa94/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L167): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L176](https://github.com/Wikia/selenium-tests/blob/3e4c31502c145b3a757f0c6d138b0ed9a9dfaa94/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L176): Introduce a new variable instead of reusing the parameter "url". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L58](https://github.com/Wikia/selenium-tests/blob/30b2486f40c91c077cc7a064822e47ebd32f01d7/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L58): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L167](https://github.com/Wikia/selenium-tests/blob/30b2486f40c91c077cc7a064822e47ebd32f01d7/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L167): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L176](https://github.com/Wikia/selenium-tests/blob/30b2486f40c91c077cc7a064822e47ebd32f01d7/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L176): Introduce a new variable instead of reusing the parameter "url". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L58](https://github.com/Wikia/selenium-tests/blob/5f408f167395eb01f9231492633ae48ada26ec38/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L58): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L167](https://github.com/Wikia/selenium-tests/blob/5f408f167395eb01f9231492633ae48ada26ec38/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L167): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L176](https://github.com/Wikia/selenium-tests/blob/5f408f167395eb01f9231492633ae48ada26ec38/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L176): Introduce a new variable instead of reusing the parameter "url". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L58](https://github.com/Wikia/selenium-tests/blob/0208e0160eb039ed52cb78f5e3adea4cd293b0e9/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L58): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L167](https://github.com/Wikia/selenium-tests/blob/0208e0160eb039ed52cb78f5e3adea4cd293b0e9/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L167): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L176](https://github.com/Wikia/selenium-tests/blob/0208e0160eb039ed52cb78f5e3adea4cd293b0e9/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L176): Introduce a new variable instead of reusing the parameter "url". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/common/dataprovider/TrackingOptInDataProvider.java [hunk 2]
```diff
@@ -310,4 +313,15 @@
         }
     };
   }
+
+  @DataProvider
+  public static Object[][] googleAnalyticAnonymizedUser() {
+    return new Object[][]{
+            {
+              Arrays.asList(
```
Comment (xkxd) [inline src/test/java/com/wikia/webdriver/common/dataprovider/TrackingOptInDataProvider.java]: could be `Collections.singletonList`

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java [hunk 3]
```diff
@@ -144,6 +144,21 @@ public void isTrackingRequestSend(List<String> elementsList,
         }
     }
 
+    public boolean areResponsesByUrlPatternSuccessful(List<String> elementsList,
+                                      NetworkTrafficInterceptor networkTrafficInterceptor) {
+        boolean result = true;
+        for(int i=0; i<elementsList.size(); i++){
```
Comment (xkxd) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java]: could be `elementsList.forEach`

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L58](https://github.com/Wikia/selenium-tests/blob/826bb8db5007e8d22e28cd4adb086eea60bb8086/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L58): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L167](https://github.com/Wikia/selenium-tests/blob/826bb8db5007e8d22e28cd4adb086eea60bb8086/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L167): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TrackingOptInModal.java#L176](https://github.com/Wikia/selenium-tests/blob/826bb8db5007e8d22e28cd4adb086eea60bb8086/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java#L176): Introduce a new variable instead of reusing the parameter "url". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**